templates/admin-page.php

```diff
@@ -45,7 +45,18 @@
 							<fieldset>
 								<legend class="screen-reader-text"><?php echo esc_html( $category ); ?></legend>
 								<label for="<?php echo esc_attr( $category ); ?>">
-									<input type="checkbox" id="<?php echo esc_attr( $category ); ?>" name="categories" value="<?php echo esc_attr( $category ); ?>" checked="checked" />
+									<?php
+										$checked = false;
+										if ( $user_settings ) {
+											$selected_categories = explode( '__', $user_settings );
+											if ( in_array( $category, $selected_categories, true ) ) {
+												$checked = true;
+											}
+										} else {
+											$checked = true;
+										}
```

Comment:
See my comment above, doing this for every category in the template isn't efficient, we should rather handle it when retrieving the user setting, so that `$user_settings` is already the array of checked categories.

Then we can simplify this here to:
```suggestion
										$checked = in_array( $category, $user_settings, true );
```

Reply:
Thank you for the review.

For the default checks, I use the default value for the `get_user_setting` function, so we can select all categories by default if the user setting is not configured.